diff --git a/share/www/script/test/rev_stemming.js b/share/www/script/test/rev_stemming.js
index 3e36f96..954da79 100644
--- a/share/www/script/test/rev_stemming.js
+++ b/share/www/script/test/rev_stemming.js
@@ -11,6 +11,9 @@
 // the License.
 
 couchTests.rev_stemming = function(debug) {
+  var db = new CouchDB("test_suite_db", {"X-Couch-Full-Commit":"false"});
+  db.deleteDb();
+  db.createDb();
   var db = new CouchDB("test_suite_db_a", {"X-Couch-Full-Commit":"false"});
   db.deleteDb();
   db.createDb();
-- 

From 37341af82ad37a2dfddc5b7e8cd4b46085101901 Mon Sep 17 00:00:00 2001
From: Wendall Cada <wendallc@apache.org>
Date: Tue, 9 Apr 2013 18:46:30 -0700
Subject: [PATCH 5/6] Add return value to waitForSuccess method.

---
 share/www/script/test/oauth.js | 1 +
 1 file changed, 1 insertion(+)

diff --git a/share/www/script/test/oauth.js b/share/www/script/test/oauth.js
index 89d3779..9199cd0 100644
--- a/share/www/script/test/oauth.js
+++ b/share/www/script/test/oauth.js
@@ -108,6 +108,7 @@ couchTests.oauth = function(debug) {
         if (xhr.responseText.indexOf("\"-pbkdf2-") != 0) {
             throw("still waiting");
         }
+        return true;
       }, "wait-for-admin");
 
       CouchDB.newUuids(2); // so we have one to make the salt
-- 
1.8.1.4


From 5e64395e08101ac090d338e9b8a9e468fd886145 Mon Sep 17 00:00:00 2001
From: Wendall Cada <wendallc@apache.org>
Date: Tue, 9 Apr 2013 18:48:08 -0700
Subject: [PATCH 6/6] Stop couchdb test server instance after javascript tests
 complete.

---
 test/javascript/run.tpl | 1 +
 1 file changed, 1 insertion(+)

diff --git a/test/javascript/run.tpl b/test/javascript/run.tpl
index 5b1e075..3736bec 100644
--- a/test/javascript/run.tpl
+++ b/test/javascript/run.tpl
@@ -118,5 +118,6 @@ else
     fi
 fi
 
+stop
 trap - 0
 exit $RESULT
-- 
1.8.1.4

