diff -up apache-couchdb-1.2.0.orig/bin/couchdb.tpl.in apache-couchdb-1.2.0/bin/couchdb.tpl.in
--- apache-couchdb-1.2.0.orig/bin/couchdb.tpl.in	2012-04-02 14:23:19.089321503 -0700
+++ apache-couchdb-1.2.0/bin/couchdb.tpl.in	2012-04-02 14:34:04.593898396 -0700
@@ -267,12 +267,21 @@ EOF
 
 stop_couchdb () {
     PID=`_get_pid`
+    STOP_TIMEOUT=60
     if test -n "$PID"; then
-        if test "$1" = "false"; then
-            echo > $PID_FILE
-        fi
+        rm -f $PID_FILE
         if kill -0 $PID 2> /dev/null; then
             if kill -1 $PID 2> /dev/null; then
+                count=0
+                while kill -0 $PID 2> /dev/null; do
+                    if [ $count -ge $STOP_TIMEOUT ]; then
+                        echo "Apache CouchDB failed to shutdown."
+                        return $SCRIPT_ERROR
+                    else
+                        count=$[count+1]
+                        sleep 1
+                    fi
+                done
                 if test "$1" = "false"; then
                     echo "Apache CouchDB has been shutdown."
                 else
