diff -up apache-couchdb-1.5.0.orig/Makefile.am apache-couchdb-1.5.0/Makefile.am
--- apache-couchdb-1.5.0.orig/Makefile.am	2013-03-11 10:26:01.394927725 -0700
+++ apache-couchdb-1.5.0/Makefile.am	2013-03-11 11:37:31.331500639 -0700
@@ -15,16 +15,16 @@
 ACLOCAL_AMFLAGS = -I m4
 
 localdoc_DATA = \
-    AUTHORS.gz \
-    BUGS.gz \
-    DEVELOPERS.gz \
-    INSTALL.gz \
-    INSTALL.Unix.gz \
-    INSTALL.Windows.gz \
-    LICENSE.gz \
-    NOTICE.gz \
-    README.gz \
-    THANKS.gz
+    AUTHORS \
+    BUGS \
+    DEVELOPERS \
+    INSTALL \
+    INSTALL.Unix \
+    INSTALL.Windows \
+    LICENSE \
+    NOTICE \
+    README.rst \
+    THANKS
 
 DISTCLEANFILES = $(localdoc_DATA)
 
@@ -32,9 +32,6 @@
     AUTHORS \
     BUGS \
     DEVELOPERS \
-    INSTALL \
-    INSTALL.Unix \
-    INSTALL.Windows \
     LICENSE \
     NOTICE \
     README.rst \
@@ -44,36 +41,6 @@
     build-aux/sphinx-touch \
     license.skip
 
-AUTHORS.gz: AUTHORS
-	gzip -9 < $< > $@
-
-BUGS.gz: BUGS
-	gzip -9 < $< > $@
-
-DEVELOPERS.gz: DEVELOPERS
-	gzip -9 < $< > $@
-
-INSTALL.gz: INSTALL
-	gzip -9 < $< > $@
-
-INSTALL.Unix.gz: INSTALL.Unix
-	gzip -9 < $< > $@
-
-INSTALL.Windows.gz: INSTALL.Windows
-	gzip -9 < $< > $@
-
-LICENSE.gz: LICENSE
-	gzip -9 < $< > $@
-
-NOTICE.gz: NOTICE
-	gzip -9 < $< > $@
-
-README.gz: README.rst
-	gzip -9 < $< > $@
-
-THANKS.gz: THANKS
-	gzip -9 < $< > $@
-
 check: dev check-js
 if TESTS
 	$(top_builddir)/test/etap/run $(top_srcdir)/test/etap
