From e724c3c7b5d23cbc1e770714a3e313a8c2c0731a Mon Sep 17 00:00:00 2001
From: Peter Lemenkov <lemenkov@gmail.com>
Date: Mon, 7 Jun 2010 15:08:42 +0400
Subject: [PATCH 07/13] Remove pid-file after stop

---
 bin/couchdb.tpl.in |    4 +---
 1 files changed, 1 insertions(+), 3 deletions(-)

diff --git a/bin/couchdb.tpl.in b/bin/couchdb.tpl.in
index 94d4743..af5cb01 100644
--- a/bin/couchdb.tpl.in
+++ b/bin/couchdb.tpl.in
@@ -262,10 +262,8 @@ EOF
 
 stop_couchdb () {
     PID=`_get_pid`
+    rm -f $PID_FILE
     if test -n "$PID"; then
-        if test "$1" = "false"; then
-            echo > $PID_FILE
-        fi
         if kill -0 $PID 2> /dev/null; then
             if kill -1 $PID 2> /dev/null; then
                 if test "$1" = "false"; then
-- 
1.7.5.1

