diff -up apache-couchdb-1.3.0.orig/Makefile.am apache-couchdb-1.3.0/Makefile.am
--- apache-couchdb-1.3.0.orig/Makefile.am	2013-03-11 10:26:01.394927725 -0700
+++ apache-couchdb-1.3.0/Makefile.am	2013-03-11 11:37:31.331500639 -0700
@@ -15,18 +15,15 @@ SUBDIRS = bin etc src share test var uti
 ACLOCAL_AMFLAGS = -I m4
 
 localdoc_DATA = \
-    AUTHORS.gz \
-    BUGS.gz \
-    CHANGES.gz \
-    DEVELOPERS.gz \
-    INSTALL.gz \
-    INSTALL.Unix.gz \
-    INSTALL.Windows.gz \
-    LICENSE.gz \
-    NEWS.gz \
-    NOTICE.gz \
-    README.gz \
-    THANKS.gz
+    AUTHORS \
+    BUGS \
+    CHANGES \
+    DEVELOPERS \
+    LICENSE \
+    NEWS \
+    NOTICE \
+    README \
+    THANKS
 
 DISTCLEANFILES = $(localdoc_DATA)
 
@@ -35,9 +32,6 @@ EXTRA_DIST = \
     BUGS \
     CHANGES \
     DEVELOPERS \
-    INSTALL \
-    INSTALL.Unix \
-    INSTALL.Windows \
     LICENSE \
     NEWS \
     NOTICE \
@@ -48,42 +42,6 @@ EXTRA_DIST = \
     build-aux/sphinx-touch \
     license.skip
 
-AUTHORS.gz: AUTHORS
-	gzip -9 < $< > $@
-
-BUGS.gz: BUGS
-	gzip -9 < $< > $@
-
-CHANGES.gz: CHANGES
-	gzip -9 < $< > $@
-
-DEVELOPERS.gz: DEVELOPERS
-	gzip -9 < $< > $@
-
-INSTALL.gz: INSTALL
-	gzip -9 < $< > $@
-
-INSTALL.Unix.gz: INSTALL.Unix
-	gzip -9 < $< > $@
-
-INSTALL.Windows.gz: INSTALL.Windows
-	gzip -9 < $< > $@
-
-LICENSE.gz: LICENSE
-	gzip -9 < $< > $@
-
-NEWS.gz: NEWS
-	gzip -9 < $< > $@
-
-NOTICE.gz: NOTICE
-	gzip -9 < $< > $@
-
-README.gz: README
-	gzip -9 < $< > $@
-
-THANKS.gz: THANKS
-	gzip -9 < $< > $@
-
 check: dev check-js
 if TESTS
 	$(top_builddir)/test/etap/run $(top_srcdir)/test/etap
