From 0f86f6e868dbdcbf832f28639bdc4c23e3304799 Mon Sep 17 00:00:00 2001
From: Peter Lemenkov <lemenkov@gmail.com>
Date: Sun, 13 Feb 2011 13:52:38 +0300
Subject: [PATCH 01/13] Do not gzip doc-files and do not install installation
 instructions

Signed-off-by: Peter Lemenkov <lemenkov@gmail.com>
---
 Makefile.am |   57 +++++++++------------------------------------------------
 1 files changed, 9 insertions(+), 48 deletions(-)

diff --git a/Makefile.am b/Makefile.am
index 6530287..dc4b8ce 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -15,18 +15,15 @@ SUBDIRS = bin etc src share test var utils
 ACLOCAL_AMFLAGS = -I m4
 
 localdoc_DATA = \
-    AUTHORS.gz \
-    BUGS.gz \
-    CHANGES.gz \
-    DEVELOPERS.gz \
-    INSTALL.gz \
-    INSTALL.Unix.gz \
-    INSTALL.Windows.gz \
-    LICENSE.gz \
-    NEWS.gz \
-    NOTICE.gz \
-    README.gz \
-    THANKS.gz
+    AUTHORS \
+    BUGS \
+    CHANGES \
+    DEVELOPERS \
+    LICENSE \
+    NEWS \
+    NOTICE \
+    README \
+    THANKS
 
 DISTCLEANFILES = $(localdoc_DATA)
 
@@ -45,42 +42,6 @@ EXTRA_DIST = \
     THANKS \
     license.skip
 
-AUTHORS.gz: $(top_srcdir)/AUTHORS
-	-gzip -9 < $< > $@
-
-BUGS.gz: $(top_srcdir)/BUGS
-	-gzip -9 < $< > $@
-
-CHANGES.gz: $(top_srcdir)/CHANGES
-	-gzip -9 < $< > $@
-
-DEVELOPERS.gz: $(top_srcdir)/DEVELOPERS
-	-gzip -9 < $< > $@
-
-INSTALL.gz: $(top_srcdir)/INSTALL
-	-gzip -9 < $< > $@
-
-INSTALL.Unix.gz: $(top_srcdir)/INSTALL.Unix
-	-gzip -9 < $< > $@
-
-INSTALL.Windows.gz: $(top_srcdir)/INSTALL.Windows
-	-gzip -9 < $< > $@
-
-LICENSE.gz: $(top_srcdir)/LICENSE
-	-gzip -9 < $< > $@
-
-NEWS.gz: $(top_srcdir)/NEWS
-	-gzip -9 < $< > $@
-
-NOTICE.gz: $(top_srcdir)/NOTICE
-	-gzip -9 < $< > $@
-
-README.gz: $(top_srcdir)/README
-	-gzip -9 < $< > $@
-
-THANKS.gz: $(top_srcdir)/THANKS
-	-gzip -9 < $< > $@
-
 check: dev
 	$(top_builddir)/test/etap/run
 
-- 
1.7.5.1

