diff -up apache-couchdb-1.5.0.orig/src/couchdb/couch_httpd_proxy.erl apache-couchdb-1.5.0/src/couchdb/couch_httpd_proxy.erl
--- apache-couchdb-1.5.0.orig/src/couchdb/couch_httpd_proxy.erl	2013-03-11 10:26:01.383927676 -0700
+++ apache-couchdb-1.5.0/src/couchdb/couch_httpd_proxy.erl	2013-03-11 10:35:36.696472496 -0700
@@ -14,7 +14,7 @@
 -export([handle_proxy_req/2]).
 
 -include("couch_db.hrl").
--include("../ibrowse/ibrowse.hrl").
+-include_lib("ibrowse/include/ibrowse.hrl").
 
 -define(TIMEOUT, infinity).
 -define(PKT_SIZE, 4096).

diff -up apache-couchdb-1.5.0.orig/src/couch_replicator/src/couch_replicator_httpc.erl apache-couchdb-1.5.0/src/couch_replicator/src/couch_replicator_httpc.erl
--- apache-couchdb-1.5.0.orig/src/couch_replicator/src/couch_replicator_httpc.erl	2013-03-11 10:26:01.387927694 -0700
+++ apache-couchdb-1.5.0/src/couch_replicator/src/couch_replicator_httpc.erl	2013-03-11 11:26:28.009602564 -0700
@@ -14,7 +14,7 @@
 
 -include("couch_db.hrl").
 -include("couch_replicator_api_wrap.hrl").
--include("../ibrowse/ibrowse.hrl").
+-include_lib("ibrowse/include/ibrowse.hrl").
 
 -export([setup/1]).
 -export([send_req/3]).

diff -up apache-couchdb-1.5.0.orig/src/couch_replicator/src/couch_replicator_utils.erl apache-couchdb-1.5.0/src/couch_replicator/src/couch_replicator_utils.erl
--- apache-couchdb-1.5.0.orig/src/couch_replicator/src/couch_replicator_utils.erl	2013-03-11 10:26:01.388927698 -0700
+++ apache-couchdb-1.5.0/src/couch_replicator/src/couch_replicator_utils.erl	2013-03-11 11:28:22.110921863 -0700
@@ -21,7 +21,7 @@
 -include("couch_db.hrl").
 -include("couch_replicator_api_wrap.hrl").
 -include("couch_replicator.hrl").
--include("../ibrowse/ibrowse.hrl").
+-include_lib("ibrowse/include/ibrowse.hrl").
 
 -import(couch_util, [
     get_value/2,
