diff -up apache-couchdb-1.5.0.orig/configure.ac apache-couchdb-1.5.0/configure.ac
--- apache-couchdb-1.5.0.orig/configure.ac	2013-03-11 09:49:50.975637942 -0700
+++ apache-couchdb-1.5.0/configure.ac	2013-03-11 09:54:34.027005943 -0700
@@ -679,7 +679,7 @@ AC_SUBST([bug_uri], ["LOCAL_BUG_URI"])
 
 AC_SUBST([localconfdir], [${sysconfdir}/${package_identifier}])
 AC_SUBST([localdatadir], [${datadir}/${package_identifier}])
-AC_SUBST([localdocdir], [${datadir}/doc/${package_identifier}])
+AC_SUBST([localdocdir], [${datadir}/doc/${package_identifier}-${version}])
 AC_SUBST([locallibdir], [${libdir}/${package_identifier}])
 AC_SUBST([localstatelibdir], [${localstatedir}/lib/${package_identifier}])
 AC_SUBST([localstatelogdir], [${localstatedir}/log/${package_identifier}])
